contracts/libraries/AccountLib.sol

```diff
@@ -228,5 +228,10 @@ function getNetBalance(
 ) view returns (int256 net) {
     uint256 internalBalance = self.reserves[token];
     uint256 physicalBalance = __balanceOf__(token, account);
+
+    // Before casting `internalBalance` into an `int256`,
+    // we must ensure it fits within. If it does not, we revert.
+    if (internalBalance > uint256(type(int256).max)) revert();
+
     net = int256(physicalBalance) - int256(internalBalance);
```

Comment:
I guess the assumption is being made here that the physical balance will never exceed `type(int256).max`...I agree that's unlikely (barring an infinite mint bug, in which case the pool is probably getting drained anyway), but it might be good to state that assumption explicitly in a comment.

Reply:
I'd still feel better if both values were checked

Reply:
Same here tbh, there do exist tokens like WETH10 that allow infinite flash mints.

Reply:
Yes, checking both values would be better, I'm fixing this right now.

Reply:
fixed